test/tests/mesh/mesh_only/tests

```diff
@@ -56,5 +56,36 @@
     method = '!dbg'
     issues = '#14312'
     requirement = 'The system shall support writing parallel checkpoint files with --mesh-only'
-  [../]
+  []
+
+  [mesh_only_with_extra_elem_ids]
+    type = 'Exodiff'
+    input = 'mesh_only_with_elem_ids.i'
+    cli_args = '--mesh-only'
+    exodiff = 'mesh_only_with_elem_ids_in.e'
+    recover = false
+    method = '!dbg'
+    issues = '#23386'
+    requirement = 'The system shall allow writing out the mesh with extra element ids without running a simulation.'
+  []
+  [mesh_only_with_restricted_elem_ids]
+    type = 'Exodiff'
+    input = 'mesh_only_with_elem_ids.i'
+    cli_args = '--mesh-only mesh_only_with_elem_ids_in.e Outputs/out/output_extra_element_ids=true Outputs/out/extra_element_ids_to_output="pin_id"'
+    exodiff = 'mesh_only_with_elem_ids_in.e'
+    recover = false
+    method = '!dbg'
+    issues = '#23386'
+    requirement = 'The system shall allow writing out the mesh with extra element ids without running a simulation and user-provided id names.'
+  []
+  [mesh_only_with_unrecognized_elem_ids]
+    type = 'Exodiff'
+    input = 'mesh_only_with_elem_ids.i'
+    cli_args = '--mesh-only mesh_only_without_elem_ids_in.e Outputs/out/output_extra_element_ids=true Outputs/out/extra_element_ids_to_output="assembly_id"'
+    exodiff = 'mesh_only_without_elem_ids_in.e'
+    recover = false
+    method = '!dbg'
+    issues = '#23386'
+    requirement = 'The system shall not write out the extra elements ids on a mesh if user provided id names do not exist on mesh.'
```

Comment:
I think it s better to have an exception and turn this test into a RunException
This is more likely to be a mispelled parameter than anything else